habitat/datasets/pointnav/__init__.py

```diff
@@ -3,3 +3,24 @@
 # Copyright (c) Facebook, Inc. and its affiliates.
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
+
+from habitat.core.registry import registry
+from habitat.core.dataset import Dataset
+
+
+def _try_register_pointnavdatasetv1():
```

Comment:
Can we make it more generic? If end user wants to add their own datasets, this is unnecessary cumbersome that needs to be followed now. A wrapper for registration in this form and documentation update for adding datasets should suffice.

Reply:
I can't imagine the situation when registration of any dataset class will fail. It will fail during instance initialization if required scenes or episode files are missing. I tend to just include `from habitat.datasets.pointnav.pointnav_dataset import PointNavDatasetV1` in `__init__.py` for registration.

Reply:
Before this try catch things were failing if a user did not have `habitat-sim` installed and only had `pyrobot` installed. This is because in our package tree we were importing actions from `habitat-sim` which was a result of having an action space in our `core` module which used `habitat-sim`. In this PR I moved that `action_space` away from `core` but we still had dependencies in our other modules like task and dataset which were still calling `habitat-sim` actions. 

This is stopgap measure, ideally I should have implemented an action space inside our `core` and made `pyrobot` and `habitat-sim` actions derive from it. Will do that in another PR and these try-imports will go away then.